.github/scripts/rust/lint.rs

```diff
@@ -0,0 +1,255 @@
+    let path = cwd.join(".cargo/config.toml");
+
+    if !path.exists() || !path.is_file() {
+        panic!(".cargo/config.toml needs to exist")
+    }
+
+    let contents = fs::read_to_string(&path).expect(&format!("should be able to read {}", path.display()));
+
+    // brute-force, search for the line:
+    // ## START CLIPPY LINTS ##
+    // and then:
+    // ## END CLIPPY LINTS ##
+    // take all values in between
+
+    if !contents.contains("## START CLIPPY LINTS ##") || !contents.contains("## END CLIPPY LINTS ##") {
+        panic!("malformed .cargo/config.toml, please add the required markers")
+    }
+
+    let file = contents.lines()
+        .map(|value| value.trim())
+        .skip_while(|value| *value != "## START CLIPPY LINTS ##")
+        .take_while(|value| *value != "## END CLIPPY LINTS ##")
+        .filter(|value| !value.is_empty())
+        .filter(|value| value.starts_with('"'))
+        // remove leading `"` and trailing `",`
+        .map(|value| &value[1..(value.len() - 2)]).fold(LintFile::default(), |mut acc, line| {
+            let (arg, lint) = line.split_at(2);
+
+            match arg {
+                "-A" => acc.allow.insert(lint.to_owned()),
+                "-W" => acc.warn.insert(lint.to_owned()),
+                "-D" => acc.deny.insert(lint.to_owned()),
+                "-F" => acc.forbid.insert(lint.to_owned()),
+                _ => panic!("unrecognized rust flag {arg}")
+            };
+
+            acc
+        });
+
+    file
+}
+
+fn apply(cwd: &Path) {
+    let path = cwd.join(".cargo/config.toml");
+
+    if !path.exists() || !path.is_file() {
+        panic!(".cargo/config.toml needs to exist")
+    }
+
+    let lints = collect_lints(cwd);
+
+    // read the `config.toml`
+    let contents = fs::read_to_string(&path).expect(&format!("should be able to read {}", path.display()));
+
+    if !contents.contains("## START CLIPPY LINTS ##") || !contents.contains("## END CLIPPY LINTS ##") {
+        panic!("malformed .cargo/config.toml, please add the required markers")
+    }
+
+    // strip out content between the markers
+    let prefix = contents.lines().take_while(|value| value.trim() != "## START CLIPPY LINTS ##");
+    let suffix = contents.lines().skip_while(|value| value.trim() != "## END CLIPPY LINTS ##");
+
+    let indent = "    ";
+    let body = lints.allow.into_iter().map(|lint| format!(r#"{indent}"-A{lint}","#))
+        .chain(lints.warn.into_iter().map(|lint| format!(r#"{indent}"-W{lint}","#)))
+        .chain(lints.deny.into_iter().map(|lint| format!(r#"{indent}"-D{lint}","#)))
+        .chain(lints.forbid.into_iter().map(|lint| format!(r#"{indent}"-F{lint}","#)));
+
+    let contents: Vec<_> = prefix.map(|line| line.to_owned())
+        .chain(once(format!("{indent}## START CLIPPY LINTS ##")))
+        .chain(body)
+        .chain(suffix.map(|line| line.to_owned()))
+        .collect();
+
+    fs::write(&path, contents.join("\n")).expect("able to write `.cargo/config.toml`");
```

Comment:
The way the `expect` function is called is wrong. This would generate an output like
> thread 'main' panicked at 'able to write `.cargo/config.toml`'

```suggestion
    fs::write(&path, contents.join("\n")).expect("unable to write `.cargo/config.toml`");
```